data/mods/ssb/abilities.ts

```diff
@@ -196,887 +472,1759 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
+			for (const target of this.getAllActive()) {
+				if (target === pokemon) continue;
+				if (target.hasAbility('turbulence')) {
+					this.field.weatherData.source = target;
+					return;
+				}
 			}
+			this.field.clearWeather();
 		},
-	},
-	// Dragontite
-	iceabsorb: {
-		desc: "This Pokemon is immune to Ice-type moves and restores 1/4 of its maximum HP, rounded down, when hit by an Ice-type move.",
-		shortDesc: "This Pokemon heals 1/4 of its max HP when hit by Ice moves; Ice immunity.",
-		name: "Ice Absorb",
+		name: "Turbulence",
 		isNonstandard: "Custom",
-		onTryHit(target, source, move) {
-			if (target !== source && move.type === 'Ice') {
-				if (!this.heal(target.baseMaxhp / 4)) {
-					this.add('-immune', target, '[from] ability: Ice Absorb');
+		gen: 8,
+	},
+
+	// cant say
+	ragequit: {
+		desc: "If a Pokemon with this ability uses a move that misses or fails, the Pokemon faints and reduces the foe's Attack and Special Attack by 2 stages",
+		shortDesc: "If move misses or fails, use Memento.",
+		name: "Rage Quit",
+		onAfterMove(pokemon, target, move) {
+			if (pokemon.moveThisTurnResult === false) {
+				this.add('-ability', pokemon, 'Rage Quit', 'boost');
+				pokemon.faint();
+				if (pokemon.side.foe.active[0]) {
+					this.boost({atk: -2, spa: -2}, pokemon.side.foe.active[0], pokemon);
 				}
-				return null;
 			}
 		},
-	},
-	// E4 Flint
-	starkmountain: {
-		desc: "The user summons Sunny Day when it switches in. In addition, the Base Power of Water-type attacks is multiplied by 0.5 against this Pokemon.",
-		shortDesc: "Summons Sunny Day on switch-in. Base Power of foe's Water moves is halved.",
-		name: "Stark Mountain",
 		isNonstandard: "Custom",
+		gen: 8,
+	},
+
+	// Celine
+	guardianarmor: {
+		desc: "On switch-in, this Pokemon's Defense and Special Defense are raised by 2 stages.",
+		name: "Guardian Armor",
 		onStart(pokemon) {
-			this.field.setWeather('sunnyday', pokemon);
+			this.boost({def: 2, spd: 2}, pokemon);
 		},
-		onSourceBasePower(basePower, attacker, defender, move) {
-			if (move.type === 'Water') {
-				return this.chainModify(0.5);
+		isNonstandard: "Custom",
+		gen: 8,
+	},
+
+	// drampa's grandpa
+	oldmanpa: {
+		desc: "This Pokemon's sound-based moves have their power multiplied by 1.3. This Pokemon takes halved damage from sound-based moves. This Pokemon ignores other Pokemon's Attack, Special Attack, and accuracy stat stages when taking damage, and ignores other Pokemon's Defense, Special Defense, and evasiveness stat stages when dealing damage. Upon switching in, this Pokemon's Defense and Special Defense are raised by 1 stage.",
+		shortDesc: "Effects of Punk Rock + Unaware. On switch-in, boosts Def and Sp. Def by 1.",
+		name: "Old Manpa",
+		onBasePowerPriority: 7,
+		onBasePower(basePower, attacker, defender, move) {
+			if (move.flags['sound']) {
+				this.debug('Old Manpa boost');
+				return this.chainModify([0x14CD, 0x1000]);
```

Comment:
What do these magic numbers mean? Can we perhaps have a comment or a `const` for them? It looks like it's 4,096 and 4,096 * 1.3, but that's quite opaque.

Reply:
This is the same as `5325/4096`, or the proper modifier in games for those technical "1.3x" abilities/moves modifiers

Reply:
Ah, okay. Thanks for explaining it!